src/services/dialog.service.ts

```diff
@@ -0,0 +1,99 @@
+import BackgroundAction from '../constants/background-actions.enum'
+import { Message } from '../messaging/scripts.messaging'
+import { DialogQuestion } from '../model/internal-messages.model'
+
+const POPUP_WIDTH = 360
+const POPUP_HEIGHT = 360
+const POPUP_TIMEOUT = 25000
+
+export class Dialog {
+  public ask(question: string, placeholders?: Record<string, string>): Promise<boolean> {
+    return new Promise<boolean>(async (resolve, reject) => {
+      try {
+        const position = this.getPosition()
+
+        const window = await chrome.windows.create({
+          url: 'dialog.html',
+          type: 'popup',
+          width: POPUP_WIDTH,
+          height: POPUP_HEIGHT,
+          ...position,
+        })
+
+        const messageHandler = this.createDialogMessageHandler(window.id, question, placeholders, resolve)
+
+        chrome.runtime.onMessage.addListener(messageHandler)
+
+        const timeout = setTimeout(() => {
```

Comment:
This timeout is added because we already have a timeout mechanism in the library. There, it's set to wait for 30 seconds, and if there is no response from the extension, request is stopped and promise rejected.

Now with this implementation, some requests might be blocked with this dialog and they can timeout because of the users inactivity. It would be inconvenient to throw an error on the library side, and the user then confirms the action and complete it. So to avoid that, I've added this timeout here, so if the user doesn't respond in some period of time the dialog is automatically canceled.

Reply:
okay, but where is this timeout function called?  : D

Reply:
The `timeout` is not a function. It's just a handle object, that is used to clear the timeout when dialog is closed. It's called just below this setTimeout:

https://github.com/fairDataSociety/blossom/blob/9ace95f70b694c3f6caf3733fbc66fc07065f249/src/services/dialog.service.ts#L32-L38